docs/migrate-to-ccm-with-csimigration.md

```diff
@@ -0,0 +1,198 @@
+
+## Before you begin
+
+Note: This guide works on a cluster created by `kubeadm`.
+
+Example `kubeadm.yaml`:
+
+```yaml
+apiVersion: kubeadm.k8s.io/v1beta2
+kind: ClusterConfiguration
+kubernetesVersion: 1.17.0-beta.1
+networking:
+  podSubnet: 10.96.0.0/16
+  serviceSubnet: 10.97.0.0/16
+controllerManager:
+  extraArgs:
+    cloud-provider: openstack
+    cloud-config: /etc/kubernetes/cloud.conf
+  extraVolumes:
+  - name: cloud
+    hostPath: /etc/kubernetes/cloud.conf
+    mountPath: /etc/kubernetes/cloud.conf
+    readOnly: true
+---
+apiVersion: kubeadm.k8s.io/v1beta2
+kind: InitConfiguration
+localAPIEndpoint:
+  bindPort: 6443
+nodeRegistration:
+  kubeletExtraArgs:
+    cloud-provider: openstack
+    cloud-config: /etc/kubernetes/cloud.conf
+```
+
+## Migrate to openstack-cloud-controller-manager
+
+Remember... In the first step, we want to keep the in-tree volume API, but for all of the rest, we want to use `openstack-cloud-controller-manager`. So we can't just remove the `--cloud-provider` flag from `kube-controller-manager`. In fact, we must disable all cloud related controllers.
+
+```yaml
+apiVersion: kubeadm.k8s.io/v1beta2
+kind: ClusterConfiguration
+kubernetesVersion: 1.17.0-beta.1
+networking:
+  podSubnet: 10.96.0.0/16
+  serviceSubnet: 10.97.0.0/16
+controllerManager:
+  extraArgs:
+    cloud-provider: openstack
+    cloud-config: /etc/kubernetes/cloud.conf
+    controllers: '*,bootstrapsigner,tokencleaner,-cloud-node-lifecycle,-route,-service'
+  extraVolumes:
+  - name: cloud
+    hostPath: /etc/kubernetes/cloud.conf
+    mountPath: /etc/kubernetes/cloud.conf
+    readOnly: true
+```
+
+And create new manifests for `kube-controller-manager`:
+
+```bash
+kubeadm init --config kubeadm-disable-controllers.yaml phase control-plane controller-manager
+```
+
+To verify, check the logs of `kube-controller-manager` for the following lines.
+
+```
+W1117 19:59:34.094182       1 controllermanager.go:513] "service" is disabled
+W1117 19:59:34.094187       1 controllermanager.go:513] "route" is disabled
+W1117 19:59:44.988353       1 controllermanager.go:513] "cloud-node-lifecycle" is disabled
+```
+
+Now you can deploy `openstack-cloud-controller-manager` using [RBAC](https://github.com/kubernetes/cloud-provider-openstack/tree/master/cluster/addons/rbac) and the [CCM manifest](https://github.com/kubernetes/cloud-provider-openstack/blob/master/manifests/controller-manager/openstack-cloud-controller-manager-ds.yaml). But disable the controller `cloud-node` by adding the argument `--controllers=*,-cloud-node`. This will be done by the `kubelets` up until the entire migration is done. Note: You must create a secret `cloud-config` with a valid `cloud.conf` for the CCM. E.g.
```

Comment:
"Now you can deploy" -> "You can now deploy". Not that the first is incorrect, but the latter flows a bit better.